packages/frontend-2/components/settings/user/profile/Details.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div class="flex flex-col gap-y-4">
+    <SettingsSectionHeader title="Your details" subheading />
+    <div class="grid md:grid-cols-2 pt-4">
+      <div class="flex h-full items-center justify-center">
+        <SettingsUserProfileEditAvatar :user="user" size="xxl" />
+      </div>
+      <div class="pt-6 md:pt-0">
+        <FormTextInput
+          v-model="name"
+          class="pt-2 pb-1"
+          color="foundation"
+          label="Name"
+          name="name"
+          placeholder="John Doe"
+          show-label
+          :rules="[isRequired, isStringOfLength({ maxLength: 512 })]"
+          @change="save"
+        />
+        <hr class="mt-4 mb-2" />
+        <FormTextInput
+          v-model="company"
+          color="foundation"
+          label="Company"
+          name="company"
+          placeholder="Example Ltd."
+          show-label
+          :rules="[isStringOfLength({ maxLength: 512 })]"
+          @change="save"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
+import { debounce } from 'lodash-es'
+import { graphql } from '~~/lib/common/generated/gql'
+import type { User, UserUpdateInput } from '~~/lib/common/generated/gql/graphql'
+import { isRequired, isStringOfLength } from '~~/lib/common/helpers/validation'
+import { useUpdateUserProfile } from '~~/lib/user/composables/management'
+
+graphql(`
+  fragment UserProfileEditDialogBio_User on User {
+    id
+    name
+    company
+    bio
+    ...UserProfileEditDialogAvatar_User
+  }
+`)
+
+const props = defineProps<{
+  user: User
```

Comment:
You're asking for way too much when requiring the full User type, making it practically impossible to actually use these components without overriding/ignoring TS errors, use fragments to define specific requirements -
https://www.notion.so/speckle/General-Web-Code-Review-Guide-1f2bf3dfea634460b3778ac3cf4aa1bb?pvs=4#d40e40efc32f4e5ea67b68e848aadcd0